src/avalara/transaction_builder_methods.py

```diff
@@ -136,42 +147,58 @@ def with_line(self, amount, quantity, item_code, tax_code, line_number=None):
         :param  string  item_code: Code of the item.
         :param  string  tax_code:  Tax Code of the item. If left blank, \
         the default item (P0000000) is assumed.
-        :param  [int]  line_number: Value of the line number.
+        :param  Any     line_number: Value of the line number.
+        :param  bool    tax_included: Is line amount tax included, defaults \
+            to false
+        :param  string    exemption_code: exemption_code in case of exempt \
+            line, defaults to blank
         :return:  TransactionBuilder
         """
+        # fix for issue #133
+        # in case line number is set as string, use it as transaction line
+        # number and continue incrementing object level line number for other
+        # lines (with no line numbers)
+        txn_line_number = self.line_num
         if line_number is not None:
-            self.line_num = line_number;
-        
+            if isinstance(
+                line_number, (int, float, complex)
+            ) and not isinstance(line_number, bool):
```

Comment:
Is the second part of this 'if' statement required? If the line_number is int, float or complex then it can't be bool right?

Reply:
bools are instances of integers in Python hence added check for bool type